exonum-java-binding/site-examples/src/test/java/com/exonum/binding/example/testing/TestkitExamplesTest.java

```diff
@@ -0,0 +1,259 @@
+      // Perform assertions on the data in the service schema
+      MapIndex<String, String> testMap = schema.testMap();
+      assertTrue(testMap.containsKey("key1"));
+      assertThat(testMap.get("key1")).isEqualTo("v1");
+    }
+  }
+
+  private static TransactionMessage createPutTransaction(int serviceId, String key, String value) {
+    return TransactionMessage.builder()
+        .serviceId(serviceId)
+        .transactionId(PUT_TX_ID)
+        .payload(
+            PutTransactionArgs.newBuilder()
+                .setKey(key)
+                .setValue(value)
+                .build())
+        .sign(TEST_KEY_PAIR);
+  }
+
+  @Test
+  void errorTransactionExecutionTest() {
+    try (TestKit testKit = TestKit.forService(ARTIFACT_ID, ARTIFACT_FILENAME,
+        SERVICE_NAME, SERVICE_ID, ARTIFACTS_DIR)) {
+      // Construct a transaction that will throw an ExecutionException
+      // with the given error code during its execution
+      byte errorCode = 100;
+      TransactionMessage validTx = createErrorTransaction(SERVICE_ID, errorCode);
+
+      // Commit block with this transaction
+      testKit.createBlockWithTransactions(validTx);
+
+      // Retrieve a snapshot of the current database state
+      BlockchainData blockchainData = testKit.getBlockchainData(SERVICE_NAME);
+      // Check that the transaction failed
+      Blockchain blockchain = blockchainData.getBlockchain();
+      Optional<ExecutionStatus> txResult = blockchain.getTxResult(validTx.hash());
+      assertThat(txResult).isPresent();
+      ExecutionStatus executionStatus = txResult.get();
+      ExecutionError executionError = executionStatus.getError();
+      assertThat(executionError.getCode()).isEqualTo(errorCode);
+      assertThat(executionError.getDescription()).isEmpty();
+      // ExecutionError also includes some properties of the error
+      // that are set by the framework: error kind, call site, etc.
+      // (not shown here, see ExecutionError API)
+    }
+  }
+
+  private TransactionMessage createErrorTransaction(int serviceId, byte errorCode) {
+    return TransactionMessage.builder()
+        .serviceId(serviceId)
+        .transactionId(RAISE_ERROR_TX_ID)
+        .payload(
+            RaiseErrorArgs.newBuilder()
+                .setCode(errorCode)
+                .build())
+        .sign(TEST_KEY_PAIR);
+  }
+
+  @Test
+  void createInPoolTransactions() {
+    try (TestKit testKit = TestKit.forService(ARTIFACT_ID, ARTIFACT_FILENAME,
+        SERVICE_NAME, SERVICE_ID, ARTIFACTS_DIR)) {
+      // Create a block so that the `afterCommit` method is invoked
+      testKit.createBlock();
+
+      // This block will contain the in-pool transactions - namely, the transaction
+      // submitted in the `afterCommit` method
+      Block block = testKit.createBlock();
+
+      // Check the resulting block or blockchain state
+    }
+  }
+
+  // code-snippet ci_timeOracleTest {
```

Comment:
maybe it would be better to have snippet-per-file?
i.e. such structure:
```
site-examples/src/test/java/com/exonum/binding/example/testing/
 testkit/
    TimeOracleExampleTest.java
    ExtensionExampleTest.java
```

Reply:
Possibly — do you propose to have a correspondence between sections and packages? On the one hand, it might make it more apparent what is included at each inclusion point: you just navigate to the file; on the other — these snippets are for a particular section and share some code (constants) — but that could be extracted.

Reply:
just suggestion. The test has 200+ lines now, not so huge, but anyway.